comp/core/secrets/secretsimpl/secrets.go

```diff
@@ -175,8 +268,13 @@ func isEnc(str string) (bool, string) {
 
 // Decrypt replaces all encrypted secrets in data by executing
 // "secret_backend_command" once if all secrets aren't present in the cache.
-func Decrypt(data []byte, origin string) ([]byte, error) {
-	if data == nil || secretBackendCommand == "" {
+func (r *secretResolver) Decrypt(data []byte, origin string) ([]byte, error) {
+	if !r.enabled {
+		e := fmt.Errorf("Agent secrets is disabled by caller")
+		log.Error(e)
+		return nil, e
+	}
```

Comment:
We should not return an error here. Comp or package should be able to call decrypt without having to worry if the main enabled the secrets feature or not.